doc/migration/v0.1.0-migration-guide.md

```diff
@@ -0,0 +1,268 @@
+
+Copy the code from the old project's `Handle()` function over the existing code in your controller's `Reconcile()` function.
+Be sure to keep the initial section in the `Reconcile()` code that looks up the object for the Request and checks to see if it's deleted.
+
+```Go
+import (
+    apierrors "k8s.io/apimachinery/pkg/api/errors"
+    cachev1alpha1 "github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1"
+    ...
+)
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the Memcached instance
+	instance := &cachev1alpha1.Memcached{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+    }
+
+    // Rest of your reconcile code goes here.
+    ...
+}
+```
+
+Change the following in your reconcile code:
+- Replace `return err` with `return reconcile.Result{}, err`
+- Replace the calls to the SDK client(Create, Update, Delete, Get, List) with the reconciler's client
+
+See the examples below and the controller-runtime [client API doc][client-api-doc] for more details.
+```Go
+// Create
+dep := &appsv1.Deployment{...}
+err := sdk.Create(dep)
+// v0.0.1
+err := r.client.Create(context.TODO(), dep)
+
+// Update
+err := sdk.Update(dep)
+// v0.0.1
+err := r.client.Update(context.TODO(), dep)
+
+// Delete
+err := sdk.Delete(dep)
+// v0.0.1
+err := r.client.Delete(context.TODO(), dep)
+
+// List
+podList := &corev1.PodList{}
+listOps := &metav1.ListOptions{LabelSelector: labelSelector}
+err := sdk.List(memcached.Namespace, podList, sdk.WithListOptions(listOps))
+// v0.1.0
+labelSelector := labels.SelectorFromSet(labelsForMemcached(memcached.Name))
+listOps := &client.ListOptions{Namespace: memcached.Namespace, LabelSelector: labelSelector}
+err := r.client.List(context.TODO(), listOps, podList)
+
+// Get
+dep := &appsv1.Deployment{APIVersion: "apps/v1", Kind: "Deployment", Name: name, Namespace: namespace}
+err := sdk.Get(dep)
+// v0.1.0
+dep := &appsv1.Deployment{}
+err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, dep)
+```
+
+Lastly copy and intialize any other fields that you may have had in your `Handler` struct into the `Reconcile<Kind>` struct:
+
+```Go
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
```

Comment:
@shawn-hurley dependency injection for the reconciler? Is that how we initialize the client with the manager's client? That's already shown as `&ReconcileMemcached{client: mgr.GetClient(), ...}`.
Or did you mean something else.

Reply:
Yeah we can go over that in a separate doc. For the migration users most likely won't need to inject fields into the controller, since the sdk handler should map directly onto the reconciler.